tests/boot/grub_test_snapshot.pm

```diff
@@ -14,28 +14,39 @@ use testapi;
 
 sub run() {
     my $self = shift;
-
     assert_screen "grub2";
     # prevent grub2 timeout; 'esc' would be cleaner, but grub2-efi falls to the menu then
+
     send_key 'up';
     if (get_var("BOOT_TO_SNAPSHOT")) {
         send_key_until_needlematch("boot-menu-snapshot", 'down', 10, 5);
         send_key 'ret';
         # find out the before migration snapshot
-        send_key_until_needlematch("snap-before-update",    'down', 40, 5) if (get_var("UPGRADE"));
-        send_key_until_needlematch("snap-before-migration", 'down', 40, 5) if (get_var("MIGRATION_ROLLBACK"));
+        send_key 'end';
+        send_key_until_needlematch("snap-before-update",    'up', 40, 5) if (get_var("UPGRADE"));
+        send_key_until_needlematch("snap-before-migration", 'up', 40, 5) if (get_var("MIGRATION_ROLLBACK"));
         send_key "ret";
+        send_key 'up';    #stop grub
+
         # bsc#956046  check if we are in first menu-entry, or not
         if (check_screen("boot-menu-snapshot-bootmenu")) {
             record_soft_failure 'bsc#956046';
             send_key 'down', 1;
             save_screenshot;
         }
-        send_key 'ret';
+
+        send_key 'e';
+        save_screenshot;
+        for my $i (1 .. 14) { sleep 1; send_key "down"; }
+        send_key 'end';
+        send_key 'spc';
+        type_string '3';
+        wait_idle;
```

Comment:
NO activity :) I've used it to make sure after that the '3' is no the correct position. I can change this to what you suggest.